tron/trondaemon.py

```diff
@@ -149,13 +156,48 @@ def __init__(self, options):
         self.context = self._build_context(options)
         self.manhole_sock = f"{self.options.working_dir}/manhole.sock"
 
+    def _make_sigint_handler(self, prev_handler=None):
+        """ Creates a SIGINT handler that takes into account a previous
+        handler to differentiate between a user request to shutdown, versus
+        another source we want to prevent from interrupting the reactor.
+
+        :type prev_handler: function
+        :param prev_handler: The previous SIGINT handler, set by another source.
+                             We use it to verify whether or not a SIGINT we
+                             received is a genuine shutdown request.
+        """
+        def handler(signum, frame):
+            try:
+                if prev_handler is not None:
+                    prev_handler(signum, frame)
+            except KeyboardInterrupt:
+                # Previous signal handler didn't raise another exception,
+                # so must be user requesting shutdown.
+                pass
+            except Exception as e:
+                # We received a SIGINT, but was caused by another thread
+                # aborting due to its own error. In this case, we don't want to
+                # stop running.
+                log.info(f"Non-reactor thread raised: {e}")
```

Comment:
I think this should be log.error. At *least* warning-level.

One thought - what do people think about disabling Mesos with MesosClusterRepository.shutdown here? It would be nice that Mesos actions would automatically start failing, instead of getting stuck in "starting". On the other hand, that assumes that these interrupts are only coming from pymesos.

Reply:
I'm not sure about disabling Mesos here precisely because, as you mentioned, we assume the interrupts come only from pymesos, @qui. In practice, that's true because its the only other place (we know of) that sets its own signal handler, but it doesn't preclude it from happening in the future (even though it really shouldn't be allowed).

Reply:
But in the future, if some other library calls interrupt_main, will they even necessarily set their own handler? I think we're already making some implicit assumptions about pymesos. (We could add more comments to make our assumptions clear.)